crates/holochain/src/conductor/api/api_external/admin_interface.rs

```diff
@@ -83,16 +106,39 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
                 // Install Dnas
                 let tasks = dnas.into_iter().map(|dna_payload| async {
                     let InstallAppDnaPayload {
-                        path,
+                        path: maybe_path,
+                        hash: maybe_hash,
                         properties,
                         membrane_proof,
                         nick,
                     } = dna_payload;
-                    let dna = read_parse_dna(path, properties).await?;
-                    let hash = dna.dna_hash().clone();
-                    let cell_id = CellId::from((hash.clone(), agent_key.clone()));
-                    self.conductor_handle.install_dna(dna).await?;
-                    ConductorApiResult::Ok((InstalledCell::new(cell_id, nick), membrane_proof))
+                    if maybe_path.is_none() && maybe_hash.is_none() {
+                        return Err(ConductorApiError::DnaReadError("Neither path nor hash specified in payload".to_string()))
+                    };
+                    if maybe_path.is_some() && maybe_hash.is_some() {
+                        return Err(ConductorApiError::DnaReadError("Both path and hash specified in payload, pick just one".to_string()))
+                    }
```

Comment:
I think this can be removed by using an enum, see my comment below on `InstallAppDnaPayload`

Reply:
I have an opinion about this as well, also below

Reply:
Because this is going to be deprecated, I think I'd just like to leave it as is.